### DISCUSSION TITLE
[error]: There was an uncaught error

### DISCUSSION THREAD

**Discussion by li5bo5** [danny-avila/LibreChat#1716] [Troubleshooting]
### What is your question?

After I deployed this project on Docker，Directly enter the web，Can be used normally，And reply normally using Gemini，
But，After stop Docker，Second launch of this project，I encountered an error message，Enclosure “error.log” can directly open it with Notepad，It is the Docker startup and operation error log that I copied。

This is the startup log：
[error.log](https://github.com/danny-avila/LibreChat/files/14151348/error.log)

2024-02-03 19:45:44 LibreChat         | 2024-02-03 11:45:44 error: KeyvMongo connection error: getaddrinfo ENOTFOUND mongodb
2024-02-03 19:45:44 LibreChat         | 2024-02-03 11:45:44 error: There was an uncaught error: getaddrinfo ENOTFOUND mongodb

### More Details

"LibreChat" and "chat-meilisearch" can start normally，
but "chat-mongodb" always starting and stopping 。
Like the video：

https://github.com/danny-avila/LibreChat/assets/6594141/9ab9aa1c-9b22-451e-a53b-e58d26f7e708

### What is the main subject of your question?

Other

### Screenshots

[error.log](https://github.com/danny-avila/LibreChat/files/14151251/error.log)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by li5bo5** [comment]
Please Help Me
tkx!

**Reply by danny-avila** [reply]
Note: MongoDB does not support older ARM CPUs like those found in Raspberry Pis. However, you can make it work by setting MongoDB’s version to mongo:4.4.18 in docker-compose.yml, the most recent version compatible with. This is likely the issue for you since WiredTiger is throwing an incompatibility error.

It is running completely fine for me after updating: 

You can try updating after you've changed the mongo version manually: 

```bash
# Stop the container (if running)
docker-compose down
# Switch to the repo's main branch
git checkout main
# Pull the latest changes to the main branch from Github
git pull 
# Prune all LibreChat Docker images
docker rmi librechat:latest
# Remove all unused dangling Docker images.
# Be careful, as this will delete all dangling docker images on your
# computer, also those not created by LibreChat!
docker image prune -f
# Building a new LibreChat image without cache
docker-compose build --no-cache

# Start LibreChat
docker-compose up
```

**Reply by fuegovic** [reply]
You should probably use the docker-compose.override.yml file to change the mongo version. There's an example override file in the librechat that shows how to do it, here's the relevant part: 

docker-compose.override.yaml:
```yaml
version: '3.4'
services:

  mongodb:
    image: mongo:4.4.18
```

once you created this file in the librechat folder, you need to run `docker-compose up -d` to apply the changes.

**Reply by danny-avila** [reply]
thanks @fuegovic !

**Comment by li5bo5** [comment]
【Reinstall “LibreChat” in "Docker-Desktop"，Same as previously described;
 "Ctrl+C" to stop the running,
"docker-compose up" to restart ,
now,error has  occurred.】

@fuegovic 
@danny-avila 
**Thank you for your responses！！**
 I have performed the operations as per your suggestions, but I am still encountering the same error.

Did I make a mistake in my operation？

 My specific actions were:
1.Ctrl+C，Exit "Windows Power Shell"
Stop "Docker Desktop" and Quit "Docker-Desktop"，
2. Open "C:\Users\Administrator\LibreChat",and I found a file named "docker-compose.yaml",
3. In the docker-compose.yaml file, at line 30, I changed "image: mongo" to "image: mongo:4.4.18"，
4. Saved and Exit.
5. Run "Docker-Desktop"
6.  I executed the following sequence of Docker Compose commands
`
cd LibreChat

docker-compose down

git checkout main

git pull 

docker rmi librechat:latest

docker image prune -f

docker-compose build --no-cache

docker-compose up -d

docker-compose up
`

7.Same error as before...

**Comment by li5bo5** [comment]
![1](https://github.com/danny-avila/LibreChat/assets/6594141/13bbf119-c894-48ae-b2f1-34fd464b8850)

**Comment by li5bo5** [comment]
![2](https://github.com/danny-avila/LibreChat/assets/6594141/8d82b02e-83ab-4556-a7eb-9da2ad12761a)

**Comment by li5bo5** [comment]
LibreChat         | 2024-02-04 06:43:43 error: KeyvMongo connection error: getaddrinfo ENOTFOUND mongodb
LibreChat         | 2024-02-04 06:43:44 error: There was an uncaught error: getaddrinfo ENOTFOUND mongodb
